### PR TITLE
Added test for Build fails completely if a URL is wikilinked in the front-matter #318

### PR DESCRIPTION
**Pull request by craignicol**
#318

## Summary by CodeRabbit

- **Tests**
  - Added new test to verify correct parsing of titles with wikilinks.
  - Introduced a system test to validate consistent output of the index generation tool.
  - Improved test file structure by fixing a missing closing brace.
  - Added tests for orphan link detection covering various link formats and edge cases.
  - Updated a wiki link test to handle trailing characters.
- **Chores**
  - Updated and added YAML front matter in test markdown files, including new author and title fields.
- **New Features**
  - Added a comprehensive markdown file for integration testing of links, orphans, contexts, tags, and tasks.
- **Documentation**
  - Expanded reference documentation with detailed links, orphans, tags, tasks, and metadata for testing purposes.
  - Updated code of conduct with simplified inline links.
  - Corrected typographical errors and improved instructions in the README.
  - Enhanced task list documentation with formatting improvements and added alternatives.
- **Bug Fixes**
  - Fixed file output handling to ensure directories are created before writing.
  - Corrected array concatenation and syntax issues for improved robustness.
- **Refactor**
  - Refined orphan link detection pattern to improve accuracy in identifying orphaned links.

### CODE REVIEW

File: src/tests/system/links-orphans-contexts-tags-tasks/expected/references.md [hunk 1]
```diff
@@ -28,8 +43,10 @@
 <details>
 <summary>Show Orphans</summary>
 
-* [Readme][README] `README.md`: [This is a link to a page that doesn't exist - an orphan]
-* [tasks][tasks] `tasks.md`: [Titles] links into [[xxxxxxxxxxxxx]]
+* [Zettel Linter][tasks] `tasks.md`: [Titles] links into [[xxxxxxxxxxxxx]], 
+* [Testing file][links] `links-orphans-contexts-tags.md`: [[https://example.com/author/bob]], [This is a link to a page that doesn't exist - an orphan], 
+* [Readme][README] `README.md`: [[This is a link to a page that doesn't exist - an orphan]], [[wiki]]`, 
+* [Contributing to the project][CONTRIBUTING] `CONTRIBUTING.md`: [TaskCollector](src/collectors/task-collector.ts) - a collector that looks for `[ ], [WikiCollector](src/collectors/wiki-collector.ts) - a collector that looks for `[[WikiLinks]]` and Zettelkasten-style `[yymmddhhmmss]`, [Templator](src/Templator.ts) uses `mustache` to render the output file, using the data collected by the collectors, and some additional annotations to support filtering notes `{{``tag[filter]`, [yyyymmddhhmmss]` links into WikiLinks `[[like-this]]`,
```
[the remark this answers is not in the document]
Comment (craignicol) [reply src/tests/system/links-orphans-contexts-tags-tasks/expected/references.md:49]: This is an expected extension. We support {{`` tags to escape the contents

File: src/tests/system/links-orphans-contexts-tags-tasks/expected/references.md [hunk 2]
```diff
@@ -39,19 +56,17 @@
...
 
-* #Recurring : [tasks][tasks]
-* #Template : [tasks][tasks]
-* #ToDo : [tasks][tasks]
-* #hashtag : [Readme][README]
-
+* #Recurring => [Zettel Linter][tasks], 
+* #Template => [Zettel Linter][tasks], 
+* #ToDo => [Zettel Linter][tasks], 
+* #hashtag => [Testing file][links], [Readme][README], [Readme][README],
```
[the remark this answers is not in the document]
Comment (craignicol) [reply src/tests/system/links-orphans-contexts-tags-tasks/expected/references.md:65]: Will be fixed in #408
